redwood-treehouse-host/build.gradle

```diff
@@ -19,6 +19,7 @@ kotlin {
         api libs.zipline
         api libs.zipline.loader
         api projects.redwoodProtocol
+        api projects.redwoodProtocolCompose
```

Comment:
I don't think this dependency is needed. And it would be very advantageous to omit it since it transitively brings in a lot of libraries that are not needed on the host side.

The `EventPublisher` implements the `ProtocolMismatchHandler` from this dependency, but it's invoked from `RealAppLifecycleHost` callbacks which are overrides of `AppLifecycle.Host` and then the implementation only forwards them to `EventListener`. So we basically trampoline through the `ProtocolMismatchHandler` for no reason and it is the only type for which this dependency was needed. It also allocates an instance which implements `ProtocolMismatchHandler` each one of the callbacks are invoked.

Instead, the Compose mismatch handler callbacks should be normal functions on `EventPublisher` which forward to its `EventListener`. This avoids the dependency and avoids the allocations.

Reply:
Tracing back my thought process... it's introduced this way, because I was following the pattern of `api projects.redwoodProtocolWidget`

I haven't formed my opinions (too new to the codebase), so definitely gonna take your suggestion.

Reply:
Yeah in general the host should only care about the widgets and the guest should only care about the composables. It's not _quite_ that clean because the composables actually also care about the widgets, but that's a bit of implementation detail and so the host=widget, guest=compose mental model is usually Good Enough™.